### PR TITLE
Fix Lobotomy

### PR DESCRIPTION
**Pull request by angelofallars**
# Description

Fixes several bugs in the lobotomy procedure.

The lobotomy effect is now stored in the brain instead of the body, so transferring a lobotomized brain will still give the lobotomy effects, and a brain transplant to a body where a lobotomy occurred no longer applies the lobotomy effects. The "Mend brain tissue" procedure to reverse a lobotomy has been unlocked after a bug prevented it from showing in the surgery UI. 

Lobotomies now add the `ClumsyComponent`, which makes the lobotomized target as clumsy as clowns.

## Technical Details

This deletes [SurgeryComponentConditionComponent.cs](https://github.com/Simple-Station/Einstein-Engines/compare/master...angelofallars:Einstein-Engines:fix-lobotomy?expand=1#diff-3786e2be1879fd877a8b501352bbd92baa3a17aecfa4a62827ad41497deb0fd7) which was only used for the lobotomy procedures (incorrectly, it was checking for `OhioAccentComponent` in the body part) in favor of [SurgeryPartComponentConditionComponent.cs](https://github.com/Simple-Station/Einstein-Engines/compare/master...angelofallars:Einstein-Engines:fix-lobotomy?expand=1#diff-7e180742b3a6f00b9f867d3ee4e8891dd00587dc4a2da8ad5e199180a387d18d) and [SurgeryBodyComponentConditionComponent.cs](https://github.com/Simple-Station/Einstein-Engines/compare/master...angelofallars:Einstein-Engines:fix-lobotomy?expand=1#diff-249e5a937ba929ffc76f85e8a43f17918afc9ba866e81f4ea4eba2c90fd0c408).

These two components are currently unused as the lobotomy procedures use a new condition component checking for the brain's `OrganComponent.OnAdd` field, but they provide a way to check for components on the body part and on the body, respectively.

## Media

**Lobotomy**
image

**Mend brain tissue**
image

**Remove organ step picture**

image

## Changelog

:cl: Skubman
- add: The lobotomy procedure makes the target clumsy like the clown. This makes them bonk when climbing tables and makes guns they're shooting blow up on their face.
- tweak: The lobotomy step now requires a scalpel instead of a drill.
- fix: Enabled the "Mend brain tissue" surgical procedure on a lobotomized target. 
- fix: The lobotomized effect is now stored in the brain instead of the body. The same brain stays lobotomized throughout brain transplants, and transferring a normal brain to a body where a lobotomy occurred no longer applies the lobotomized effect.
- fix: The lobotomy procedure now shows the proper popup during the lobotomization step.
- fix: Removed the ability to perform lobotomies on bodies without a brain.
- fix: The "Remove organ" surgery step on the UI now properly shows the retractor sprite instead of the hemostat.

### CODE REVIEW

File: Content.Shared/Medical/Surgery/SharedSurgerySystem.Steps.cs [hunk 1]
```diff
@@ -120,6 +121,48 @@ private void OnToolStep(Entity<SurgeryStepComponent> ent, ref SurgeryStepEvent a
             }
         }
 
+        if (ent.Comp.AddOrganOnAdd != null)
```
Comment (sleepyyapril) [inline Content.Shared/Medical/Surgery/SharedSurgerySystem.Steps.cs:124]: I really don't like this

File: Content.Shared/Medical/Surgery/SharedSurgerySystem.Steps.cs [hunk 2]
```diff
@@ -182,6 +225,55 @@ private void OnToolCheck(Entity<SurgeryStepComponent> ent, ref SurgeryStepComple
...
+                }
+
+                foreach (var key in compsToAdd.Keys)
+                {
+                    if (!organ.OnAdd.ContainsKey(key))
+                    {
+                        args.Cancelled = true;
+                        return;
+                    }
+                }
```
Comment (sleepyyapril) [inline Content.Shared/Medical/Surgery/SharedSurgerySystem.Steps.cs:251]: despair. please clean this up, you could very easily remove most of the curlies

**Review by sleepyyapril — CHANGES_REQUESTED** [review]

File: Content.Shared/Medical/Surgery/SharedSurgerySystem.Steps.cs (hunk 1 above)
Comment (sleepyyapril) [reply Content.Shared/Medical/Surgery/SharedSurgerySystem.Steps.cs:124]: same reason as below, you could do this a lot better I think.

**Comment by sleepyyapril** [comment]
I'll accept this when you try to murder the curlies as best you can.

**Comment by angelofallars** [comment]
Tried to murder the curlies as best as I can, but some of them might still be alive though.

**Review by sleepyyapril — APPROVED** [review]

**Review by VMSolidus — APPROVED** [review]

**Pull request merged**